src/orch/orch.c

```diff
@@ -1,7 +1,47 @@
+#include "../common/dbus.h"
+#include "controller.h"
+
 #include <stdio.h>
 #include <stdlib.h>
 
-int main() {
+int getArgs(int argc, char *argv[], int *ctrlPort) {
+	if (argc < 2) {
+		fprintf(stderr, "No controller port given\n");
+		return -1;
+	}
+	*ctrlPort = atoi(argv[1]);
```

Comment:
As mentioned in the systemd coding standards, we shouldn't use plain atoi().  Instead, we can define a safe version of atoi like systemd does, in src/basic/parse-util.c:
`int safe_atoi(const char *s, int *ret_i)`

Reply:
Good point. I tried to implement a safer version via [strtol](https://man7.org/linux/man-pages/man3/strtol.3.html), which seemed pretty straight forward and allows better error detection. Not sure about disadvantages. Please have a look at `src/common/parse-util.h`